packages/module/src/Message/CodeBlockMessage/CodeBlockMessage.tsx

```diff
@@ -5,25 +5,94 @@ import { useState, useRef, useId, useCallback, useEffect } from 'react';
 import SyntaxHighlighter from 'react-syntax-highlighter';
 import { obsidian } from 'react-syntax-highlighter/dist/esm/styles/hljs';
 // Import PatternFly components
-import { CodeBlock, CodeBlockAction, CodeBlockCode, Button, Tooltip } from '@patternfly/react-core';
+import {
+  CodeBlock,
+  CodeBlockAction,
+  CodeBlockCode,
+  Button,
+  Tooltip,
+  ExpandableSection,
+  ExpandableSectionToggle,
+  ExpandableSectionProps,
+  ExpandableSectionToggleProps
+} from '@patternfly/react-core';
 
 import { CheckIcon } from '@patternfly/react-icons/dist/esm/icons/check-icon';
 import { CopyIcon } from '@patternfly/react-icons/dist/esm/icons/copy-icon';
-import { ExtraProps } from 'react-markdown';
+
+export interface CodeBlockProps {
+  /** Content rendered in code block */
+  children?: React.ReactNode;
+  /** Aria label applied to code block */
+  'aria-label'?: string;
+  /** Class name applied to code block */
+  className?: string;
+  /** Whether code block is expandable */
+  isExpandable?: boolean;
+  /** Length of text initially shown in expandable code block; defaults to 10 characters */
+  maxLength?: number;
+  /** Additional props passed to expandable section if isExpandable is applied */
+  expandableSectionProps?: Omit<ExpandableSectionProps, 'ref'>;
+  /** Additional props passed to expandable toggle if isExpandable is applied */
+  expandableSectionToggleProps?: ExpandableSectionToggleProps;
+  /** Link text applied to expandable toggle when expanded */
+  expandedText?: string;
+  /** Link text applied to expandable toggle when collapsed */
+  collapsedText?: string;
+}
 
 const CodeBlockMessage = ({
   children,
   className,
   'aria-label': ariaLabel,
+  isExpandable = false,
+  maxLength = 10,
+  expandableSectionProps,
+  expandableSectionToggleProps,
+  expandedText = 'Show less',
+  collapsedText = 'Show more',
   ...props
-}: Omit<JSX.IntrinsicElements['code'], 'ref'> & ExtraProps) => {
+}: CodeBlockProps) => {
   const [copied, setCopied] = useState(false);
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [initialWidth, setInitialWidth] = useState<number | null>(null);
 
-  const buttonRef = useRef(undefined);
+  const buttonRef = useRef();
   const tooltipID = useId();
+  const toggleId = useId();
+  const contentId = useId();
+  const codeBlockRef = useRef<HTMLDivElement>(null);
+
+  // Keeps the width the same - this can vary when content is expandable
+  useEffect(() => {
+    if (codeBlockRef.current) {
+      let ancestor = codeBlockRef.current.parentNode;
+      if (ancestor) {
+        ancestor = ancestor.parentNode;
+        if (ancestor) {
+          ancestor = ancestor.parentNode;
+          if (ancestor && ancestor instanceof HTMLElement) {
+            setInitialWidth(ancestor.offsetWidth);
```

Comment:
A comment of which ancestor element this chain is targeting may be helpful to have in case of any HTML updates.